challenge-20/challenge-20.js

```diff
@@ -1,107 +1,136 @@
+  resultado à uma variável chamada `username`. Se o usuário não digitar um
+  nome, `username` deve receber "Desconhecido".
+  Com a resposta, mostre um alert com a mensagem "Bem vindo [USERNAME]!"
+  */
+  var username = prompt( 'Qual o seu nome?' ) || 'Desconhecido';
+  console.log( username );
+
+  /*
+  Agora, pergunte ao usuário "Qual o seu e-mail?", atribuindo o resultado à
+  uma variável chamada `email`.
+  */
+  var email = prompt( 'Qual seu email?' );
+
+  /*
+  - Selecione o input de "Nome", atribuindo-o à uma variável chamada
+  `$inputUsername`.
+  */
+  var $inputUsername = doc.querySelector( '[type="text"]' );
+  console.log( $inputUsername );
+  /*
+  - Selecione o input de "Email", atribuindo-o à uma variável chamada
+  `$inputEmail`.
+  */
+  var $inputEmail = doc.querySelector( '[type="email"]' );
+  console.log( $inputEmail );
+
+  /*
+  - Selecione o campo de "Mensagem", atribuindo-o à uma variável chamada
+  `$message`.
+  */
+  var $message = doc.getElementsByTagName( 'textarea' )[0];
+  console.log( $message );
+
+  /*
+  - Selecione o botão de envio do formulário, atribuindo-o à uma variável
+  chamada `$button`.
+  */
+  var $button = doc.querySelector( '[type="submit"]' );
+  console.log( $button );
+
+  /*
+  Preencha os campos de "Nome" e "Email" que estão no documento com os valores
+  entrados pelo usuário.
+  */
+  $inputUsername.value = username;
+  $inputEmail.value = email;
+
+  /*
+  Adicione um listener de evento de click ao botão que faça o seguinte:
+  1. Verificar se todos os campos estão preenchidos:
+  - Mostrar um alert para cada campo não preenchido, como abaixo:
+  - Se o campo de "Nome" não estiver preenchido, mostrar:
+      - "Preencha o nome do usuário!"
+  - Se o campo de "Email" não estiver preenchido, mostrar:
+      - "Preencha o e-mail!"
+  - Se o campo de "Mensagem" não estiver preenchido, mostrar:
+      - "Preencha a mensagem!"
+  - Se o campo de "Email" for inválido, mostrar:
+      - "Entre com um e-mail válido!"
+
+  2. Para verificar se o e-mail é válido use a função `isValidEmail`, passando
+  o e-mail que foi entrado no campo de "Email" por parâmetro. (A função
+  `isValidEmail` será criada logo abaixo).
+
+  3. Se tudo estiver OK, pergunte ao usuário:
+      - "Tem certeza que deseja enviar o formulário?"
+  Se for confirmado, mostre um alerta com a mensagem:
+      - "Enviado com sucesso!"
+  Caso contrário, mostre um alerta com a mensagem:
+      - "Não enviado."
+  */
+  $button.addEventListener( 'click', function( event ) {
+    event.preventDefault();
+
+    if( $inputUsername.value === '' )
+      alert( 'Preencha o nome do usuário!' );
+    else if( $inputEmail.value === '' )
+      alert( 'Preencha o e-mail!' );
+    else if( !(isValidEmail( $inputEmail.value )) )
+      alert( 'Entre com um e-mail válido!' );
+    else if( $message.value === '' )
+      alert( 'Preencha a mensagem!' );
+    else
+      confirm( 'Tem certeza que deseja enviar o formulário?' ) ?
```

Comment:
-1
Só use ternário quando for atribuir o valor resultante do ternário, ou quando for retornar valores de funções.